sound/soc/intel/boards/sof_sdw.c

```diff
@@ -468,7 +473,7 @@ static int get_sdw_dailink_info(const struct snd_soc_acpi_link_adr *links,
 		u64 adr;
 
 		adr = link->adr_d->adr;
-		codec_index = find_codec_info_part(adr);
+		codec_index = find_codec_info_part(adr, link->adr_d->dai_name);
```

Comment:
I see the same being done in this function elsewhere too, so maybe it needs to be fixed for all cases. `link->adr_d` is an array, right? That's what its description says. So `link->adr_d->x` should actually be `link->adr_d[0].x`. And in this function we use the first element in the array because there's only one element or because the data fields that we're interested in is the same in all elements? Is this "obvious" for anyone really familiar with SDW or would we need a comment here?

Reply:
> I see the same being done in this function elsewhere too, so maybe it needs to be fixed for all cases. `link->adr_d` is an array, right? That's what its description says. So `link->adr_d->x` should actually be `link->adr_d[0].x`. And in this function we use the first element in the array because there's only one element or because the data fields that we're interested in is the same in all elements? Is this "obvious" for anyone really familiar with SDW or would we need a comment here?

You are right. It is a kind of limitation of the existing code. Currently, we can only support the same codecs in a snd_soc_acpi_adr_device array. For example, we can support rt1308 + rt1308, but can't support rt1316 + rt1308 on the same sdw link. It could be something we can enhance in the future.